application/src/test/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/mappers/RaptorRequestMapperTest.java

```diff
@@ -91,19 +113,77 @@ void testPassThroughPoints() {
   }
 
   @Test
-  void testPassThroughPointsTurnTransitGroupPriorityOff() {
+  void testTransitGroupPriority() {
     var req = new RouteRequest();
 
-    // Set pass-through and relax transit-group-priority
-    req.setViaLocations(List.of(new PassThroughViaLocation("Via A", List.of(STOP_A.getId()))));
+    // Set relax transit-group-priority
     req.withPreferences(p ->
       p.withTransit(t -> t.withRelaxTransitGroupPriority(CostLinearFunction.of("30m + 1.2t")))
     );
 
     var result = map(req);
 
-    //  transit-group-priority CANNOT be used with pass-through and is turned off...
-    assertTrue(result.multiCriteria().transitPriorityCalculator().isEmpty());
+    assertTrue(result.multiCriteria().transitPriorityCalculator().isPresent());
+  }
+
+  static List<Arguments> testViaAndTransitGroupPriorityCombinationsTestCases() {
+    return List.of(
+      Arguments.of(List.of(VIA_VISIT), List.of(VIA_VISIT), null),
+      Arguments.of(List.of(VIA_PASS_THROUGH), List.of(VIA_PASS_THROUGH), null),
+      Arguments.of(List.of(TRANSIT_GROUP_PRIORITY), List.of(TRANSIT_GROUP_PRIORITY), null),
+      Arguments.of(List.of(RELAX_COST_DEST), List.of(RELAX_COST_DEST), null),
+      Arguments.of(
+        List.of(VIA_VISIT, TRANSIT_GROUP_PRIORITY),
+        List.of(VIA_VISIT, TRANSIT_GROUP_PRIORITY),
+        null
+      ),
+      Arguments.of(
+        List.of(VIA_PASS_THROUGH, TRANSIT_GROUP_PRIORITY, RELAX_COST_DEST),
+        List.of(VIA_PASS_THROUGH),
+        null
+      ),
+      Arguments.of(
+        List.of(VIA_VISIT, TRANSIT_GROUP_PRIORITY, RELAX_COST_DEST),
+        List.of(VIA_VISIT, TRANSIT_GROUP_PRIORITY),
+        null
+      ),
+      Arguments.of(
+        List.of(TRANSIT_GROUP_PRIORITY, RELAX_COST_DEST),
+        List.of(TRANSIT_GROUP_PRIORITY),
+        null
+      ),
+      Arguments.of(
+        List.of(VIA_VISIT, VIA_PASS_THROUGH),
+        List.of(),
+        "A mix of via-locations and pass-through is not allowed in this version."
+      )
+    );
```

Comment:
What was the logic for what combination of features are tested for?

Would it be useful to have a test for all possible combinations, so that the current functionality is documented and any functionality changes result in failed tests?

Reply:
Only 4 features can not be used in combinations witch exclude other features. The business rules are not clear from this, so I will add those - hopefully the reason for not adding more test-cases become clear. The features override each other in this order: VIA_PASS_THROUGH -> VIA_VISIT -> TRANSIT_GROUP_PRIORITY -> RELAX_COST_DEST. So, testing all combination does not add any value. I added more doc and a few test cases.